src/Generic/Data/Internal/Enum.hs

```diff
@@ -63,6 +84,27 @@ class GEnum opts f where
 -- report](https://www.haskell.org/onlinereport/haskell2010/haskellch11.html#x18-18400011.2)).
 data StandardEnum
 
+-- | Extends the 'StandardEnum' option for 'GEnum' to allow all constructors to 
+-- have arbitrary many arguments. Each argument type/field must be an instance 
+-- of both 'Enum' and 'Bounded'. Two restrictions require the user's caution:
+--
+-- * The instances of the field types need to be valid. Particularly 'Int' is an 
+-- unfit field type, because the enumeration of the negative values starts 
+-- before 0. 
```

Comment:
We should make explicit what fields are "valid" (a counterexample like `Int` is still fine to illustrate though). I think we can say that they're instances of `Enum` and `Bounded`, and that the enumeration starts from 0.